### DISCUSSION TITLE
`hideOnInteractOutside` doesn't work when <Menu /> is rendered in shadow DOM

### DISCUSSION THREAD

**Discussion by crisshaker** [ariakit/ariakit#3970] [General]
When a Menu is rendered in a Shadow DOM, clicking outside when the menu is open doesn't close the menu.

Is there a good-enough workaround for this, at least? 

This is a live reproduction of this behavior: https://ariakit-shadow-dom-issue.vercel.app/
The source code: https://github.com/crisshaker/ariakit-shadow-dom-issue

Thanks!!!

**Comment by crisshaker** [comment]
Also another weird behavior I noticed is that the ```autoSelect="always"``` prop doesn't work as expected! It doesn't auto select the first non-disabled element whenever the value/list changes

cc @diegohaz

**Comment by benrodrs** [comment]
I think it's safe to say we don't support shadow DOM at the moment, as we don't have any tests for this scenario. You can manually hide the popover in response to an interaction outside the shadow root and implement your own `autoSelect` logic, but we can't guarantee other features will work.
